### PR TITLE
Play nicely with message forwarding

### PR DESCRIPTION
**Pull request by gf3**
* Indicate whether a value can properly respond to a message

### CODE REVIEW

**Comment by tomstuart** [comment]
Sorry, I had notifications turned off, so I only just noticed this.

This overlaps with #2, but your implementation is different. Which one is right?

File: spec/monads/optional_spec.rb [hunk 1]
```diff
@@ -94,6 +94,10 @@ module Monads
       it 'returns the message’s result wrapped in an Optional' do
         expect(optional.challenge.value).to eq response
       end
+
+      it 'correctly indicated whether a value can respond to a message' do
+        expect(optional.respond_to? :challenge).to be true
```
Comment (tomstuart) [inline spec/monads/optional_spec.rb]: I would rather see `expect(optional).to respond_to :challenge` here, assuming that works.

Comment (gf3) [inline spec/monads/optional_spec.rb]: oh dang, i didn't even know there was a matcher for that. will update

**Comment by gf3** [comment]
@tomstuart updated the test. i don't think the `within` usage is necessary if we want this to play well with other ruby objects/libraries. of course it's up to you

**Comment by tomstuart** [comment]
@gf3 Thanks. What would the corresponding tests look like for `Many` and `Eventually`?

**Comment by gf3** [comment]
@tomstuart ahh, i suppose we'd have to apply the `respond_to?` check to `values` and `block` respectively. alternatively we could use a common name for the stored value in each class.

**Comment by tomstuart** [comment]
@gf3 Well, that’s what `within` is for. :)

**Comment by gf3** [comment]
@tomstuart oh my mistake—i thought it also re-wrapped values.

**Comment by gf3** [comment]
@tomstuart the other PR is correct then

**Comment by tomstuart** [comment]
@gf3 It does — I’m being a bit flippant. I just mean that this implementation needs to work across all monads, so it’s going to have to use `within` (or similar) somehow.

**Comment by tomstuart** [comment]
This change is superseded by #10, so I’m closing it.

**Pull request closed without merging**